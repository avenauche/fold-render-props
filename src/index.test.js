import React from 'react'
import renderer from 'react-test-renderer'
import folder from './index'

const TEST_RENDER_PROP = 'testRenderProp'

const ComponentA = props => {
  return (props.children || props.render || props[TEST_RENDER_PROP])({
    name: props.name.toUpperCase()
  })
}

ComponentA.displayName = 'A'

const ComponentB = props => {
  return (props.children || props.render || props[TEST_RENDER_PROP])({
    name: props.name.big()
  })
}

ComponentB.displayName = 'B'

const ComponentC = props => {
  return (props.children || props.render || props[TEST_RENDER_PROP])({
    name: props.name.repeat(3)
  })
}
ComponentC.displayName = 'C'

describe('fold-render-props', () => {
  const propNames = ['render', 'children', TEST_RENDER_PROP]
  propNames.forEach(renderProp => {
    const Folder = folder(
      [
        (result, render) =>
          React.createElement(ComponentA, {
            name: '⒜' + result.name + '⒜',
            [renderProp]: render
          }),
        (result, render) =>
          React.createElement(ComponentB, {
            name: '⒝' + result.name + '⒝',
            [renderProp]: render
          }),
        (result, render) =>
          React.createElement(ComponentC, {
            name: '⒞' + result.name + '⒞',
            [renderProp]: render
          })
      ],
      {
        renderPropName: renderProp
      }
    )

    test('basic ' + renderProp, () => {
      const tree = renderer
        .create(
          React.createElement(Folder, {
            name: renderProp,
            [renderProp]: result => {
              return <pre>{JSON.stringify(result, null, 2)}</pre>
            }
          })
        )
        .toJSON()
      expect(tree).toMatchSnapshot()
    })
  })
})
