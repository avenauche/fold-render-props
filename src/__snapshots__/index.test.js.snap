// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fold-render-props basic children 1`] = `
<pre>
  {
  "name": "⒞&lt;big&gt;⒝⒜CHILDREN⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜CHILDREN⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜CHILDREN⒜⒝&lt;/big&gt;⒞"
}
</pre>
`;

exports[`fold-render-props basic render 1`] = `
<pre>
  {
  "name": "⒞&lt;big&gt;⒝⒜RENDER⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜RENDER⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜RENDER⒜⒝&lt;/big&gt;⒞"
}
</pre>
`;

exports[`fold-render-props basic testRenderProp 1`] = `
<pre>
  {
  "name": "⒞&lt;big&gt;⒝⒜TESTRENDERPROP⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜TESTRENDERPROP⒜⒝&lt;/big&gt;⒞⒞&lt;big&gt;⒝⒜TESTRENDERPROP⒜⒝&lt;/big&gt;⒞"
}
</pre>
`;
